src/main/java/com/playmonumenta/networkrelay/RemotePlayerManager.java

```diff
@@ -0,0 +1,347 @@
+package com.playmonumenta.networkrelay;
+
+import com.google.gson.JsonObject;
+import java.util.Map;
+import java.util.Objects;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.ConcurrentSkipListMap;
+import org.bukkit.Bukkit;
+import org.bukkit.entity.Player;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.EventPriority;
+import org.bukkit.event.Listener;
+import org.bukkit.event.player.PlayerJoinEvent;
+import org.bukkit.event.player.PlayerQuitEvent;
+import org.checkerframework.checker.nullness.qual.MonotonicNonNull;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public class RemotePlayerManager implements Listener {
+	public static final String REMOTE_PLAYER_CHANNEL_BASE = "monumenta.redissync.remote_player";
+	public static final String REMOTE_PLAYER_UPDATE_CHANNEL = REMOTE_PLAYER_CHANNEL_BASE + ".update";
+	public static final String REMOTE_PLAYER_REFRESH_CHANNEL = REMOTE_PLAYER_CHANNEL_BASE + ".refresh";
+
+	@MonotonicNonNull
+	private static RemotePlayerManager INSTANCE = null;
+
+	private static final Map<String, Map<String, RemotePlayer>> mRemotePlayerShardMapped = new ConcurrentSkipListMap<>();
+	private static final Map<UUID, RemotePlayer> mRemotePlayersByUuid = new ConcurrentSkipListMap<>();
+	private static final Map<String, RemotePlayer> mRemotePlayersByName = new ConcurrentSkipListMap<>();
+
+
+	private final NetworkRelay mPlugin;
+	protected RemotePlayerManager(NetworkRelay plugin) {
+        mPlugin = plugin;
+		String lShard = getShardName();
+		try {
+			for (String shard: NetworkRelayAPI.getOnlineShardNames()) {
+				if (shard.equals(lShard)) {
+					continue;
+				}
+				mRemotePlayerShardMapped.put(shard, new ConcurrentSkipListMap<>());
+			}
+		} catch (Exception ex) {
+			mPlugin.getLogger().warning("Failed to get load shards");
+		}
+
+		try {
+			NetworkRelayAPI.sendBroadcastMessage(REMOTE_PLAYER_REFRESH_CHANNEL, new JsonObject());
+		} catch (Exception e) {
+			mPlugin.getLogger().warning("Failed to broadcast to channel " + REMOTE_PLAYER_REFRESH_CHANNEL);
+        }
+
+		INSTANCE = this;
+    }
```

Comment:
Weird indentation - maybe tabs/spaces issues?

Reply:
don't believe so, no idea where it's coming from, but might be fixed in next commit